lib/Doctrine/Migrations/Generator/FileBuilder.php

```diff
@@ -65,6 +46,9 @@ public function buildMigrationFile(
 
     private function getVersionUpdateQuery(string $version, string $direction) : string
     {
+        // @todo this is hard to implement since MetadataStorage abstracts the migrations table
```

Comment:
but without this, things are broken as the metadata storage is not updated with executing this file. So this must be solved.
It might mean that the MetadataStorage needs to expose a method returning the needed queries.

Reply:
Does it make sense then to have metadata storage interface? If it is going to be only one implementation, what about dropping the interface and typehint the concrete implementation?

Reply:
well, the metadata storage interface could have a method returning the queries necessary to update the storage (which may be no queries if the storage cannot be updated in SQL).

Reply:
But then metadata storage interface will be bound to SQL, having other "storages"  will make no sense... right? or in that case the extra method for the queries will just return NULL? )

Reply:
I have discussed this with @alcaeus . This will be covered in a separate PR/ticket and there can be explored a solution for the issue